pkg/monitor/cluster/debugpods.go

```diff
@@ -0,0 +1,69 @@
+package cluster
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+
+	"github.com/Azure/ARO-RP/pkg/util/stringutils"
+)
+
+// Each cluster is being checked every 1 minute. emitDebugPodsCount
+// lists all events (age < 70 seconds) in the default namespace
+// in order to detect recent events of debug-pod creation.
+func (mon *Monitor) emitDebugPodsCount(ctx context.Context) error {
+	events, err := mon.cli.CoreV1().Events("default").List(ctx, metav1.ListOptions{
+		FieldSelector: fields.SelectorFromSet(map[string]string{
+			"reason":              "Started",
+			"involvedObject.kind": "Pod",
+			// oc debug node creates a pod with a "container-00"
+			// container name.
+			"involvedObject.fieldPath": "spec.containers{container-00}",
+		}).String(),
+	})
+	if err != nil {
+		return err
+	}
+
+	nodes, err := mon.cli.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
```

Comment:
this monitor and `emitNodeConditions` both (and possibly more) list nodes on the cluster, duplicating these calls. I think that should be refactored so that we do these calls only once. However, probably not in the scope of your PR

Reply:
@rolandmkunkel Yeah, I have thought about this kind of duplicate API calls. I guess we should balance the code readability and the performance here. Could you please provide an example of how you see the refactoring here?

Reply:
The thing is called informer https://aly.arriqaaq.com/kubernetes-informers/ and I believe it should be able to watch events. 

More resources:
- https://www.cncf.io/blog/2019/10/15/extend-kubernetes-via-a-shared-informer/
- https://pkg.go.dev/k8s.io/client-go/informers